web/src/OverviewLogPane.tsx

```diff
@@ -320,27 +372,150 @@ export class OverviewLogComponent extends Component<OverviewLogComponentProps> {
       }
       return true
     })
+
     this.logCheckpoint = patch.checkpoint
+    lines.forEach((line) => this.lineHashList.append(line))
+
+    if (startCheckpoint) {
+      // If this is an incremental render, put the lines in the forward buffer.
+      lines.forEach((line) => {
+        this.forwardBuffer.push(line)
+      })
+    } else {
+      // If this is the first render, put the lines in the backward buffer, so
+      // that the last lines get rendered first.
+      lines.forEach((line) => {
+        this.backwardBuffer.push(line)
+      })
+    }
+
+    if (this.renderBufferRafId) return
```

Comment:
what's going on here? can you explain at minimum to me, but maybe also via comment for the next reader?

Reply:
Schedule a render job if there's not one already scheduled.